### PR TITLE
Feat: Add replication support

### PR DESCRIPTION
**Pull request by anton-chekanov**
Add support for allowing replication of a database and creating a database out of available replica.

## Test Plan

* [x] acceptance tests for database and replica
* [x] unit tests for database

## References
* https://docs.snowflake.com/en/user-guide/database-replication-config.html
* https://docs.snowflake.com/en/sql-reference/sql/alter-database.html#database-replication-and-failover-syntax
* https://docs.snowflake.com/en/sql-reference/sql/show-replication-databases.html

### CODE REVIEW

File: pkg/resources/database.go [hunk 1]
```diff
@@ -147,13 +196,105 @@ func ReadDatabase(d *schema.ResourceData, meta interface{}) error {
...
+	return d.Set("replication_is_primary", replication.IsPrimary.Bool)
 }
 
 func UpdateDatabase(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	database := d.Get("name").(string)
+
+	// Change the replication details first - this is a special case and won't work using the generic method
+	x := func(resource, replType string, toggleGrant func(db *sql.DB, database, account, replType, toggleType string) error) error {
+		o, n := d.GetChange(resource)
```
Comment (alldoami) [inline pkg/resources/database.go:237]: Can you use more descriptive names for these variables?

File: pkg/resources/database_acceptance_test.go [hunk 2]
```diff
@@ -8,43 +8,84 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
+func testCheckReplication(t *testing.T, path string, replicas, failovers []string) func(*terraform.State) error {
```
Comment (alldoami) [inline pkg/resources/database_acceptance_test.go:14]: Nit: Use uppercase for test `TestCheckReplication`

Comment (anton-chekanov) [reply pkg/resources/database_acceptance_test.go]: Changed.

File: pkg/resources/database.go (hunk 1 above)
Comment (anton-chekanov) [reply pkg/resources/database.go]: Well, this is how it is in other resources, but sure, changed to old_ver/new_ver.

**Comment by alldoami** [comment]
@anton-chekanov are you still working on this? Please reopen if you are going to fix the conflicts.

**Pull request closed without merging**